.github/workflows/build_mingw.yml

```diff
@@ -0,0 +1,134 @@
+---
+name: Windows / MinGW / CPU tests
+
+permissions:
+  contents: read
+
+on:
+  push:
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - 'doc/**'
+      - 'CHANGES'
+      - 'COPYING'
+      - 'CREDITS'
+      - 'LICENSE'
+      - 'README.*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  mingw_test_matrix:
+    name: LLVM ${{ matrix.llvm_version }} - MinGW ${{ matrix.config }}
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        llvm_version: [18, 19]
+        config: [cpu]
+
+    steps:
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: MINGW64
+          update: true
+          install: >-
+            mingw-w64-x86_64-gcc
+            mingw-w64-x86_64-cmake
+            mingw-w64-x86_64-ninja
+            mingw-w64-x86_64-hwloc
+            mingw-w64-x86_64-pkg-config
+
+      - name: Checkout PoCL
+        uses: actions/checkout@v4
+
+      - name: Setup Build Environment
+        run: |
+          # Convert the Windows workspace path to an MSYS2 compatible one
+          WORKSPACE=$(cygpath -u "${{ github.workspace }}")
+          echo "WORKSPACE=${WORKSPACE}" >> $GITHUB_ENV
+
+      - name: Checkout LLVM
+        id: checkout-llvm
+        uses: actions/checkout@v4
+        with:
+          repository: llvm/llvm-project
+          ref: release/${{ matrix.llvm_version }}.x
+          path: llvm-project
+
+      - name: Restore Cached LLVM
+        id: cache-llvm
+        uses: actions/cache@v4
+        with:
+          path: install-llvm
+          key: llvm-mingw-${{ matrix.llvm_version }}-${{ runner.os }}-${{ steps.checkout-llvm.outputs.commit }}
```

Comment:
FYI, I have a [patch](https://github.com/linehill/pocl/commit/af72d61941a84537cf47a491f391e5458c7f5a77#diff-8d02397b540d39d6c1091936dbc8cbab6f4db603c27091f92c90d7afff118cdbR28) for downloading LLVM binaries for Windows instead of building it. You could use it to avoid the rebuild penalty. A downside is that the full LLVM binary with libraries and llvm-config seems to be only available since LLVM-19.

Reply:
I need a build with LLVM_BUILD_LLVM_DYLIB=ON and LLVM_LINK_LLVM_DYLIB=ON not to run into export ordinal limits, and I don't think those official binaries will use LLVM_LINK_LLVM_DYLIB...